homeassistant/components/geniushub/__init__.py

```diff
@@ -57,17 +66,20 @@
 V1_API_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_TOKEN): cv.string,
-        vol.Required(CONF_MAC): vol.Match(MAC_ADDRESS_REGEXP),
+        vol.Required(CONF_MAC): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
     }
 )
+
+
 V3_API_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_HOST): cv.string,
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_MAC): vol.Match(MAC_ADDRESS_REGEXP),
+        vol.Optional(CONF_MAC): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
```

Comment:
Why is the mac address optional?

Reply:
This is per the documentation. Using the local API the MAC address can be retrieved from the hub. If, for some reason, you can override the MAC address. I have no idea why anyone would want to do this unless they got a new hub or got it wrong when setting up the cloud API.